app/assets/javascripts/cart.js

```diff
@@ -137,20 +125,57 @@ $(document).ready(function() {
         updateCart();
     });
 
+    $(".cart-span").on("click", ".del", function() {
+        var items = JSON.parse(localStorage.getmyshop_shoppping_cart_items);
+        var element_id = $(this).attr("id");
+        var cost = 0;
+        if (items.length == 1) {
+            items[0] = undefined;
+            location.reload();
+            Materialize.toast("Your cart has been emptied!", 3000);
+        } else {
+            for (var i in items) {
+                    if (items.hasOwnProperty(i)) {
+                    cost += (1 * items[i].quantity) * (1 * items[i].price);
+                    if (parseInt(element_id) == items[i].product_id) {
+                        var parent =
+                            document.getElementById("updated_product"),
+                            stringProductId = "" + items[i].product_id + "",
```

Comment:
True @andela-echigbo, toString() was proving problematic. I found out you
need to wrap the number in parentheses to make it work. :)

On Mon, Feb 15, 2016 at 9:57 AM, Emmanuel Chigbo <notifications@github.com>
wrote:

> In app/assets/javascripts/cart.js
> <https://github.com/andela/getmyshop/pull/44#discussion_r52872896>:
>
> > +    $(".cart-span").on("click", ".del", function() {
> > +        var items = JSON.parse(localStorage.getmyshop_shoppping_cart_items);
> > +        var element_id = $(this).attr("id");
> > +        var cost = 0;
> > +        if (items.length == 1) {
> > +            items[0] = undefined;
> > +            location.reload();
> > +            Materialize.toast("Your cart has been emptied!", 3000);
> > +        } else {
> > +            for (var i in items) {
> > +                    if (items.hasOwnProperty(i)) {
> > +                    cost += (1 * items[i].quantity) * (1 * items[i].price);
> > +                    if (parseInt(element_id) == items[i].product_id) {
> > +                        var parent =
> > +                            document.getElementById("updated_product"),
> > +                            stringProductId = "" + items[i].product_id + "",
>
> You can use the javascript toString() method here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/andela/getmyshop/pull/44/files#r52872896>.
>



-- 
*Best Regards,*

*Yusuf Daniju /** Andela.com
<http://andela.com/?utm_source=andela&utm_medium=email&utm_campaign=signature> *
*Software Developer / **+234.807.266.7248 <+234.807.266.7248>** / @Andela
<http://bit.ly/andelatwitter>*

*"The startup that's harder to get into than Harvard" - *CNN
<http://snip.ly/HfId>
See what The Wall Street Journal <http://snip.ly/u9AI> has to say.